File: 890-lemonade-change/lemonade-change.cpp
class Solution {
public:
    bool lemonadeChange(vector<int>& bills) {
        int n = bills.size();
        int cntfive = 0 , cntten = 0 , cnttwo = 0;
        int i = 0;

        while(i < n ){
            if(bills[i] == 5){
                cntfive++;
            }
            else if(bills[i] == 10 ){
                if(cntfive == 0)return false;
                cntten++;
                cntfive--;
            }
            else{
                

                if(cntfive > 0 && cntten > 0){
                    cntfive--;
                    cntten--;
                }
                else if(cntfive >= 3){
                    cntfive -= 3;
                }
                else{
                    return false;
                }
            }
            i++;
        }
        if(i == n)return true;
        return false;
        
    }
};

File: 890-lemonade-change/lemonade-change.java
class Solution {
    public boolean lemonadeChange(int[] bills) {
        Integer n = bills.length;
        Integer cfive = 0 , cten = 0 , ctwo = 0;

        for(Integer i = 0 ; i < n ; i++){
            if(bills[i] == 5){
                cfive++;
            }
            else if(bills[i] == 10){
                if(cfive == 0)return false;
                else{
                    cfive--;
                    cten++;
                }
            }
            else{
                if(cten > 0 && cfive > 0){
                    ctwo++;
                    cten--;
                    cfive--;
                }
                else if(cfive > 2){
                    cfive -= 3;
                    ctwo++;
                }
                else{
                    return false;
                }
            }

        }
        return true;
        
    }
}